File: Leetcode/PY/Leetcode26-删除排序数组中的重复项.py
from typing import List
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        fast,slow = 0,0
        n = len(nums)
        while fast < n:
            if nums[fast] == nums[slow]:
                fast += 1
            else:
                slow += 1 
                nums[slow] = nums[fast]
                fast += 1
        return slow + 1
                

File: Leetcode/sbgo/Leetcode26-删除排序数组中的重复项.go
package sbgo

func removeDuplicates(nums []int) int {
	slow, fast := 0, 0
	n := len(nums)
	if n == 0 {
		return 0
	}
	for fast < n {
		if nums[fast] == nums[slow] {
			fast++
		} else {
			slow++
			nums[fast], nums[slow] = nums[slow], nums[fast]
			fast++
		}
	}
	nums = nums[:slow+1]
	return slow + 1
}
